promq/autocomplete/earley/luthor.go

```diff
@@ -275,15 +274,15 @@ func mapParserItemTypeToTokhanType(item promql.Item) TokenType {
 
 // need to explicitly extract this function since it's private
 // in prometheus 2.16
-func isAggregator(item promql.ItemType) bool {
-	return item > 57386 && item < 57398
+func isAggregator(item parser.ItemType) bool {
+	return item > 57388 && item < 57401
 }
 
-func isOperator(item promql.ItemType) bool {
-	return item > 57367 && item < 57385
+func isOperator(item parser.ItemType) bool {
+	return item == 57346 || (item > 57367 && item < 57387)
```

Comment:
The lexer will still return `EQL` for the `=` character, but the ascii code for that token is not changed to 57346: https://github.com/prometheus/prometheus/blob/main/promql/parser/generated_parser.y.go#L37

In the old version, the code for the token was 57370: https://github.com/prometheus/prometheus/blame/a6ee1f85179fdc2a9939a77807ade736a597dae6/promql/parser/generated_parser.y.go#L60

Reply:
would it make sense to use `parser.EQL` instead?

Reply:
Alternatively, it may be worth trying to upstream prometheus would be fine with exporting these special markers `OperatorsStart | End`

Reply:
> would it make sense to use `parser.EQL` instead? and `item >= parser.ADD && item <= parser.ATAN2`

yeah i like it!

Reply:
Good idea, changed